server.js

```diff
@@ -1,16 +1,121 @@
 const express = require('express');
 const helmet = require('helmet');
 const cors = require('cors');
+const mongoose = require('mongoose');
 
 const server = express();
 
 server.use(helmet());
 server.use(cors());
 server.use(express.json());
 
+mongoose
+  .connect('mongodb://localhost/Friends')
+  .then((connect) => {
+    console.log('API Server Connected');
+  })
+  .catch((error) => {
+    console.error('Db connection failed');
+  });
+
 server.get('/', (req, res) => {
   res.status(200).json({ api: 'running' });
 });
 
+server.post('/api/friends', (req, res) => {
+  const { firstName, lastName, age, createdOn } = req.body;
+  createdOn = !createdOn ? new Date() : createdOn; // initialize createdOn to current date/time if it is undefined
+  if (!firstName || !lastName || !age) {
+    res.status(400).json({ errorMessage: 'Please provide firstName, lastName, and age for the friend.' });
+  }
+  if (typeof age !== Number || age < 1 || age > 120) {
+    res.status(400).json({ errorMessage: 'Age must be a number between 1 and 120.' });
+  }
+  const friend = new Friend({ firstName, lastName, age, createdOn });
+  friend.save()
+    .then((newFriend) => {
+      res.status(201).json(newFriend);
+    })
+    .catch((error) => {
+      res.status(500).json({ errorMessage: 'There was an error while saving the friend to the database.' });
+    });
+});
+
+server.get('api/friends', (req, res) => {
```

Comment:
It's great that you're combining both `GET` routes into one! ☺️ 

Just make sure your path has `/api/friends/:id?`. The `:id?` is crucial, it makes the `req.params.id` optional.

Also, don't forget the `/` before `api/friends/:id?`